### PR TITLE
add documentation for POST /config/reload

### PR DESCRIPTION
**Pull request by cpanato**
Implemention code PR: https://github.com/mattermost/platform/pull/5713

### CODE REVIEW

File: source/v4/system.yaml [hunk 1]
```diff
@@ -14,3 +14,21 @@
             $ref: "#/definitions/StatusOK"
         '400':
           $ref: '#/responses/BadRequest'
+  '/config/reload':
+    post:
+      tags:
+        - system
+      summary: Reload the Server configuration
+      description: |
+        Reload the server configuration
```
Comment (jwilander) [inline source/v4/system.yaml]: Can the summary be `Reload configuration` and the description be `Reload the configuration file to pick up on any changes made to it.`

File: source/v4/system.yaml [hunk 2]
```diff
@@ -14,3 +14,21 @@
...
+        ##### Permissions
+        Must be authenticated and have the `PERMISSION_MANAGE_SYSTEM` permission.
+      responses:
+        '200':
+          description: Configuration successful reloaded
```
Comment (jwilander) [inline source/v4/system.yaml]: And this `Configuration reload successful`

**Comment by jwilander** [comment]
Going to need a rebase when the other system ones go in I figure

**Comment by cpanato** [comment]
@jwilander done

**Pull request merged**